### DISCUSSION TITLE
external speaker info and solution.

### DISCUSSION THREAD

**Discussion by gartnl** [egzumer/uv-k5-firmware-custom#267] [Show and tell]
So I noticed that whenever I connect an external speaker, the audio-amplifier shuts down after a few seconds.
The same happend with a baofeng speaker mike. It happens on both QS 5R's.
This got me thinking: what if the amp is meant for high impedance earplugs with DC connected output?
So I placed a 100 uF capacitor in the line and behold: sound!
Did QS really save on an output capacitor ??

**Comment by egzumer** [comment]
I have no idea how that works, however a friend reported that his earplug speaker gets hot while it is plugged in, looks like there is some DC voltage not filtered out. It is also worth noting that there are different board versions, most common are older 1.4 and newest 1.6. The newest one has some speaker-mic issues fixed (EDIT: this is not true, read more below), mainly PTT not working while receiving incoming signal, maybe the issue you mention is fixed too.

**Reply by egzumer** [reply]
Well, it is not fixed in v1.6 just checked, that is a bummer

**Reply by egzumer** [reply]
Actually this is something they broke in v1.6. There is a 3.6V DC voltage at the speaker pins, this is not the case with v1.4

**Reply by gartnl** [reply]
Strange, my speakemike isn't working. Even with the capacitor mod. The sound is there now, but no TX with the speakermike. Not even when on the other VFO. Well, they are cheap so can't complain too much.

**Reply by egzumer** [reply]
They messed that up, they used wrong, double ended audio aplifier, now they have problems, baofeng for instance uses single ended one. They fixed the PTT problem but now there is this DC component.

**Reply by OE1MWW** [reply]
yes, can confirm. UV-K5(8) on the speaker output (tip of 2.5mm) are ~3,8 volt, on the UV-K5R was no voltage at that pin.
There is a mod info floating around: the wire from speaker to ground inside the mic should be replaced by a 47uF capacitor.  
Works, no difference in sound quality

**Reply by ErikS-web** [reply]
What speaker ? Intern radio, or externally in microphone ?

**Comment by egzumer** [comment]
This is caused by a short in the radio. Now when removed the PTT doesn't work correctly again.

![20231214_173107](https://github.com/egzumer/uv-k5-firmware-custom/assets/14902414/486033e0-7838-4c04-870e-6c295b5c26be)

**Comment by ErikS-web** [comment]
![dualplug](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/3c4b2ffe-6447-4904-9a62-2e4c721208af)

When I plug in my external microfone, the TX/Red-light is on for a split second (rev 1.6 board).

Any photo on adding an elco internally. There is not much space direct at soldering point of speaker,

**Comment by ErikS-web** [comment]
Indeed a short... here's my radio. 

![Rev-1p6](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/8c9f11a9-c010-4a89-8ca0-76dee0ee83b2)

**Reply by gartnl** [reply]
How strange. They solved a problem with a short, a blob of solder in fact?

**Reply by egzumer** [reply]
This is not a fix. The cure is worse then the disease. This is an accident that it fixes anything. The speaker-mic socket is hand soldered, and contacts where the blob is are extremly close.

**Comment by OE1MWW** [comment]
![2023-12-14_180323](https://github.com/egzumer/uv-k5-firmware-custom/assets/73303392/d1d2a4e0-6ae0-47f2-969f-9d13c9d33325)

**Reply by ErikS-web** [reply]
So the modification in in externally speaker/microphone. Not intern the Radio ?

**Reply by OE1MWW** [reply]
yes, it's done in the external mic. 
![290589285-3c4b2ffe-6447-4904-9a62-2e4c721208af](https://github.com/egzumer/uv-k5-firmware-custom/assets/73303392/c4195485-935a-4bd3-bc2a-b9d2d7f72a78)

**Reply by gartnl** [reply]
Ah. So others came to the same conclusion. I used it in a communications speaker and in the external mike. The C does not hinder other sets from beeing used.

**Reply by ErikS-web** [reply]
Clear picture #thumpsup

**Comment by ErikS-web** [comment]
On SP+ I measured on my radio 3.6 volts (UV-K5(8) rev 1.6).

With 47uF I was not satisfied with the sound on low-frequenties, so I placed **100uF** in the microphone (type **Baofeng K-780MIC**).

I used an NC strip as new soldering pad for the black wire, from where the elco is positioned to the original connection (GND)

![Mic without extra elco](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/d8c01115-9f06-40e1-99db-562441769a41)

![Mic with 100uF elco](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/915ab24f-c7de-4454-ba96-58d25595aa58)

Now you can also see that the PTT2 is not connected. I you wish to have it, all it needs is a soldering connecting to PTT

**Reply by ErikS-web** [reply]
That external PTT was not working while receiving incoming signal, I had on my old rev 1,4 board.
And I thought they fixed that with 1.6... _with bad soldering ?_

I mentioned the [difference in PTT on the wiki ](https://github.com/egzumer/uv-k5-firmware-custom/wiki/Button-functions#ptt-1)

**Comment by egzumer** [comment]
There is a unpopulated place for capacitor, this capacitor fixes the issue with external PTT not working when speaker is playing sound. This is the same fix as a capacitor in speaker-mic.

Before, the path between pads has to be cut:
![Screen Shot 12-15-23 at 01 04 AM](https://github.com/egzumer/uv-k5-firmware-custom/assets/14902414/e9e30118-35e6-4c7e-a271-54570f9ce936)

After:
![290689351-1a9ecfa8-c194-46e6-a773-d7228e0a8f7e](https://github.com/egzumer/uv-k5-firmware-custom/assets/14902414/28a4b929-6dec-455e-9da6-607c55804a8d)

**Reply by OE1MWW** [reply]
any video link please - how to open - and not to break a UV-K5(8) 

:-)

**Reply by egzumer** [reply]
https://www.youtube.com/watch?v=T4LiJeMK_xI

An LCD has tabs holding it that go into a board. The first one is on the left bottom corner. It has to be pushed in to the right and pryed out. Then the second tab can be wiggled out, and LCD opens up like a flap.

**Reply by egzumer** [reply]
> Are the 4V DC present only on the negative speaker output (VO1)

No, on both.

> wouldn't that cause the 4V DC to appear on the differential output of the internal speaker

It would, it wouldn't, it doesn't matter, all at the same time. DC is not a "DC" in a transient state, speaker works on AC, DC on internal speaker has nowhere to flow either way.

**Reply by egzumer** [reply]
There is no heat without flow.

**Reply by egzumer** [reply]
No, it wouldn't. Capacitor doesn't remove offset, and DC can't flow through capacitor, so how would it flow through 16Ω speaker?

**Reply by FrancoisDeveraux** [reply]
Thank you sincerely for your help in everything here. My question: In the two photos (before, after) with the addition of the 47uF Capacitor, is it correct that there is a trace from the "-" to the "+" solder pads which must first be cut (scraped or otherwise disconnected)? I think I see it in the photo (before pic) but I am asking to be certain. My understanding is the capacitor is tantalum; must it be tantalum or would any 47uF that fits there work?

**Reply by egzumer** [reply]
Yes it has to be cut, otherwise it would just short the capacitor and it would have no effect whatsoever. I don't know if there is any other type of capcitor that would fit there, but I don't see a reason why it has to strictly be the tantalum type. I guess ceramic would be just fine too, although I don't know if there are ones with 47u or more capacity.

**Reply by ag6ju** [reply]
Hi my name is NIKO ,  AG6JU,  I worked for com-spec.com  Communications Specialists in Orange California.  owner of com-spec.com  want to hire you to write specialized UV-K5 firmware for him,  please contact  me , if you are interested.  ag6ju.2014@gmail.com

**Comment by ErikS-web** [comment]
Tonight I have three radios, revision 1.6, opening on my desk. All three have the internal short circuit, as previously reported.

Here are a few photos, before, during and after the treatment.
It is quite difficult to get to the specific place.
Raising the display is best.
Then you can easily reach it with a desoldering iron and remove the excess solder.

![Often seen Soldering short on rev 1 6 board](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/19c34cad-a169-49cc-8ab5-597a070bae86)


You can also clearly see in the middle photo that the solder blob shorted/bridged the above capacitor.
This causes DC current to flow to the connector, causing an additional load on the amplifier when an external speaker/microphone/earplug is connected

There is only 1 mm space between the solder pad and the capacitor, so a closure is easily made.
Happened to me too on my 1st soldering attempt. The solder just went there and went wrong again.
It is best to connect the soldering iron with a point the width of the pin to the other and apply some solder and let it flow.

**Reply by ErikS-web** [reply]
When this desoldering is done, a new problem occurs...

The external PTT is acting the same as on Rev 1,4 board.  "**_External PTT does NOT functions , as long a signal is received_**". 

**Solution is the same; adding the capacitor in the external microphone.**  

See the schematic and my photo's above.

**Comment by ErikS-web** [comment]
Egzumer placed a nice thermal video on Telegram 

https://t.me/quansheng_uvk5_en_dev/22/59770

Showing what happens when the short is made (with a screw driver).

Also interesting to see that the resistor (470 Ohm) just below, is heated in normal situation.

**Comment by ErikS-web** [comment]
Also interesting that you mentioned **_blank space_** on the PCB, for the 227 capacitor (220uF).

I'll look for or order some parts.

**Comment by gartnl** [comment]
Sometimes topics like this lead to a surprising discovery!
After putting the C in the speakermike TX would still not work, but sound was ok. But my UV-B5 wouldn't transmit either anymore. Turned out that in the mike the wires from the tip and sleeve of the 2.5 mm jack were reversed. 
Without the C the PTT would find a dc-path through the speaker, but with the C that path was blocked. Changing the wires solved it for both the UV-B5 and UV-5R. As a bonus the mike signal improved, because that signal also used to go through the speaker.
So much for el-cheapo mikes from the far-east ..

**Comment by ErikS-web** [comment]
Have a look at YouTube _open case uvk5_
